src/main/java/org/folio/circulation/domain/ConfigurationService.java

```diff
@@ -76,7 +76,7 @@ private boolean isConfigurationEmptyOrUnavailable(JsonObject configJson) {
   }
 
   private Integer findTimeoutDuration(JsonObject configJson) {
-    return configJson.getInteger(CHECKOUT_TIMEOUT_DURATION_KEY, DEFAULT_CHECKOUT_TIMEOUT_DURATION_IN_MINUTES);
+    return Integer.parseInt(configJson.getValue(CHECKOUT_TIMEOUT_DURATION_KEY, DEFAULT_CHECKOUT_TIMEOUT_DURATION_IN_MINUTES).toString());
```

Comment:
Could this use `JsonPropertyFetcher.getIntegerProperty`?

Reply:
Actually, `JsonPropertyFetcher.getIntegerProperty` doesn't help us, because, implementation  method `getIntegerProperty` casts the value to Number.

And if we send request with string value - we will throw exception: `java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Number`.  And it will be the same behavior as previous implementation.

Reply:
> Actually, `JsonPropertyFetcher.getIntegerProperty` doesn't help us, because, implementation method `getIntegerProperty` casts the value to Number.
> 
> And if we send request with string value - we will throw exception: `java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Number`. And it will be the same behavior as previous implementation.

That does seem to be a limitation in the vert.x implementation. 

`Integer.parseInt` [throws](https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#parseInt(java.lang.String)) a `NumberFormatException` if the string does not contain a parseable integer. Is this exception handled differently to a `ClassCastException`?

Reply:
Good point, now I handle situation when we can  throws a `NumberFormatException`. Also if we have this case I will return default value and avoid `throws` any exception.